pkg/monitoring/mimir/service.go

```diff
@@ -0,0 +1,144 @@
+package mimir
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/giantswarm/observability-operator/pkg/common"
+	"github.com/giantswarm/observability-operator/pkg/common/password"
+	"github.com/giantswarm/observability-operator/pkg/common/secret"
+	"github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent"
+)
+
+const (
+	ingressAuthSecretName = "mimir-gateway-ingress-auth"
+	mimirApiKey           = "mimir-basic-auth"
+	mimirNamespace        = "mimir"
+)
+
+type MimirService struct {
+	client.Client
+	PasswordManager password.Manager
+	common.ManagementCluster
+}
+
+// ConfigureMimir configures the ingress and its authentication (basic auth)
+// to allow prometheus agents to send their data to Mimir
+func (ms *MimirService) ConfigureMimir(ctx context.Context) error {
+	logger := log.FromContext(ctx).WithValues("cluster", ms.ManagementCluster.Name)
```

Comment:
Could we add the mc name to the context instead? That way we do not have to pass the logger around to subfunctions

Reply:
Also, I'm not sure the mc name brings a lot of value so you can probably use `log.FromContext(ctx)` directly

Reply:
@QuantumEnigmaa Can you check this one real quick? Then we're good :)

Reply:
> Could we add the mc name to the context instead?

I'm not sure how to do that though :/

Reply:
```go
bg := context.Background()
ctx := context.WithValue(bg, "k", "zabba")
```

Reply:
So you can do this at the top of the controller I think?
ctx = context.WithValue(ctx, "cluster", cluster.GetName())
ctx = context.WithValue(ctx, "installation", mcName)

and then we should have this in all log lines. Would be nice to test :)

Reply:
So this had to be done in the controller just before calling the `ConfigureMimir` function then right ?

Reply:
I think at the top of the reconcile function even :)

Reply:
Did you have time to check this one out?

Reply:
No my bad I totally forgot about this

Reply:
Done